core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -71,12 +83,77 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    static final class Fields {
-        static final String _INDEX = "_index";
-        static final String _TYPE = "_type";
-        static final String _ID = "_id";
-        static final String STATUS = "status";
-        static final String ERROR = "error";
+    public static BulkItemResponse fromXContent(XContentParser parser, int id) throws IOException {
+        XContentParser.Token token = parser.nextToken();
```

Comment:
Nit: in other places advancing the parser is directly done in the "ensureExpectedToken" call. Saves one line and I think it is equally readable. There's several places in this method where this might be possible, not sure how much lines this saves. Also, if you don't like it, just a matter of tase I think